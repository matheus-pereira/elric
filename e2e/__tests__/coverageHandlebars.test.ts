/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import * as path from 'path';
import {readFileSync} from 'graceful-fs';
import wrap from 'elric-snapshot-serializer-raw';
import {cleanup, runYarnInstall} from '../Utils';
import runelric from '../runelric';

const dir = path.resolve(__dirname, '../coverage-handlebars');
const coverageDir = path.join(dir, 'coverage');

beforeAll(() => {
  cleanup(coverageDir);
});

it('code coverage for Handlebars', () => {
  runYarnInstall(dir);
  const result = runelric(dir, ['--coverage', '--no-cache']);

  expect(result.exitCode).toBe(0);
  expect(wrap(result.stdout)).toMatchSnapshot();

  const coverageMapFile = path.join(coverageDir, 'coverage-final.json');
  const coverageMap = JSON.parse(readFileSync(coverageMapFile, 'utf-8'));

  expect(
    Object.keys(coverageMap).map(filename => path.basename(filename)),
  ).toEqual(['greet.hbs']);
});
