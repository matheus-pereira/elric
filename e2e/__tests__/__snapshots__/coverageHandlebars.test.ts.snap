// elric Snapshot v1, https://goo.gl/fbAQLP

exports[`code coverage for Handlebars 1`] = `
-----------|---------|----------|---------|---------|-------------------
File       | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
-----------|---------|----------|---------|---------|-------------------
All files  |      75 |    61.53 |   66.66 |   66.66 |                   
 greet.hbs |      75 |    61.53 |   66.66 |   66.66 | 10                
-----------|---------|----------|---------|---------|-------------------
`;
